### PR TITLE
code update to Dimension methods

### PR DESCRIPTION
**Pull request by SSaishruthi**
Code updates for:

1. __radd__
2. __rsub__
3. __rmul__

### CODE REVIEW

File: tensorflow/python/framework/tensor_shape.py [hunk 1]
```diff
@@ -337,7 +337,14 @@ def __radd__(self, other):
     Returns:
       A Dimension whose value is the sum of `self` and `other`.
     """
-    return self + other
+    try:
+      other = as_dimension(other)
+    except (TypeError, ValueError):
+      return NotImplemented
```
Comment (alextp) [inline tensorflow/python/framework/tensor_shape.py]: Why return NotImplemented? Why swallow the exception?

I have no idea from reading this PR what issue it's trying to fix and why you believe it fixes it. Can you add documentation and a test which fails before but passes after this change?

Also note that Dimension is deprecated and will go away with tfv2, so probably not much to be gained in improving it.

Comment (qlzh727) [reply tensorflow/python/framework/tensor_shape.py]: That's actually the same comment I gave on the other PR https://github.com/tensorflow/tensorflow/pull/27962#event-2301569662

Comment (SSaishruthi) [reply tensorflow/python/framework/tensor_shape.py]: Thanks, @alextp, and @qlzh727 for the review. Will update on this.

**Comment by SSaishruthi** [comment]
Since Dimension is getting deprecated. I will focus on TensorShape class. Creating new PR with changed results.

**Comment by SSaishruthi** [comment]
Since Dimension is getting deprecated. I will focus on TensorShape class. Creating new PR with changed results.

**Pull request closed without merging**